Automattic/jetpack
# Instant Search: Update prototype with overlay design

<img width="1119" alt="Screen Shot 2020-01-31 at 5 04 14 PM" src="https://user-images.githubusercontent.com/4044428/73582738-c6658780-444b-11ea-85fd-b5feaa00d15a.png">

#### Changes proposed in this Pull Request:
- New, overlay-oriented design
- Improved integration with the Customizer (plus dark/light themes)
- Added tests, bugfixes, photon support, and accessibility improvements

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Yes, this is part of the Jetpack Instant Search project: p10gg3-9iP-p2. It has not been publically launched.

#### Testing instructions:
1. Follow [these setup instructions](https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md#testing-instructions) to set up Jetpack Instant Search on your site. Please note that the Jetpack Search widget will need to be added to twice: once to your site sidebar/footer and once to the Jetpack Search Sidebar within the search overlay.

2. Submit a query into your site search input. Ensure that a search overlay appears with your search results.

3. Enter your site's customizer (`/wp-admin/customize.php`) and navigate to the "Jetpack Search" section. Ensure that changing the settings there take effect on the overlay without a page reload.

4. Try enabling and using filters on the widget in your site's sidebar/footer. Ensure that they behave as expected.

5. Try enabling and using filters on the widget inside the search overlay. Ensure that they behave as expected.

#### Proposed changelog entry for your changes:
* Implements an overlay-based design for Jetpack Instant Search.
